### PR TITLE
[user_accounts] Avoid erasing current sites from Database when saving my_preference page

### PR DESCRIPTION
**Pull request by ridz1208**
This pull request adds a validation on the user_accounts page being used to make sure sites are only processed when editing user

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
@ridz1208 

Tested this and it does indeed fix the issue on the my_preferences page.

**Comment by ridz1208** [comment]
@driusan please review and merge, this is blocking testing

File: modules/user_accounts/php/NDB_Form_user_accounts.class.inc [hunk 1]
```diff
@@ -221,24 +221,25 @@ class NDB_Form_User_Accounts extends NDB_Form
...
-            $DB->delete('user_psc_rel', array("UserID" => $uid));
-            foreach ($us_curr_sites as $site) {
-                $DB->insert(
-                    'user_psc_rel',
-                    array(
-                     "UserID"   => $uid,
-                     "CenterID" => $site,
-                    )
-                );
+        if ($this->page === 'edit_user') {
```
Comment (jstirling91) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: move this to the if block on line 366

Comment (ridz1208) [inline modules/user_accounts/php/NDB_Form_user_accounts.class.inc]: no

**Pull request merged**